lib/devise_two_factor/models/two_factor_authenticatable.rb

```diff
@@ -61,17 +56,44 @@ def clean_up_passwords
 
     protected
 
-      # An OTP cannot be used more than once in a given timestep
-      # Storing timestep of last valid OTP is sufficient to satisfy this requirement
-      def consume_otp!
-        if self.consumed_timestep != current_otp_timestep
-          self.consumed_timestep = current_otp_timestep
+      # Consumes an OTP code and returns true if successful
+      # @param [String] code the OTP code to check against
+      # @return [Boolean] true if the code wasn't already used and a more recent code hasn't been used yet.
+      #                   false if the code is invalid, as already been used or a more recent code has been consumed.
+      def consume_otp!(code, totp = otp)
+        self.consumed_timestep ||= 0 # TODO REMOVE only necessary to pass some tests
```

Comment:
I would leave it in there for this bugfix / minor version upgrade, because the [devise_two_factor_generator.rb!](https://github.com/tinfoil/devise-two-factor/blob/master/lib/generators/devise_two_factor/devise_two_factor_generator.rb#L19) does not have 0 as default value for the consumed_timestep. It will most likely lead to issues if devise_two_factor is updated without setting the consumed_timestep's default to 0 and run the migration again. 
For the next major version I recommend to set the consumed_timestep's default and remove the line.